### PR TITLE
DATA-1524 - Initial round of Data API cleanups

### PR DESCRIPTION
**Pull request by agreenb**
Part 1 of several API cleanup PRs.  Corresponding app PR: https://github.com/viamrobotics/app/pull/1934.

All of the following fields have just been removed ([1](https://github.com/viamrobotics/rdk/pull/2570), [2](https://github.com/viamrobotics/rdk/pull/2572)) or were previously unused in RDK; confirmed by testing these changes locally against RDK.

- Remove deletion result status from Deletion APIs, as we no longer have partial successes; we either error or delete everything requested.
- Remove deprecated component model. This was removed in RDK https://github.com/viamrobotics/rdk/pull/2570 and is now only used in .ts files in app due to linting errors, which will now be removed in the corresponding app PR.
- Remove deprecated tags field in the Filter, which has been replaced with the TagsFilter field.
- Remove deprecated session ID field. This was removed in RDK https://github.com/viamrobotics/rdk/pull/2574 and the corresponding app PR.
- Remove TabularCapture, BinaryCapture, and InternalPackage, which are unused and have been replaced with internal Golang structs in app.
- Remove app/data/v1/model.proto, which was replaced in app and RDK by packages.proto months ago.

### CODE REVIEW

File: proto/viam/app/datasync/v1/data_sync.proto
Comment (agreenb) [inline proto/viam/app/datasync/v1/data_sync.proto:79]: Unused in RDK.

File: proto/viam/app/datasync/v1/data_sync.proto
Comment (agreenb) [inline proto/viam/app/datasync/v1/data_sync.proto:88]: Replaced in app months ago with [dbBinaryData](https://github.com/viamrobotics/app/blob/4e313dba532bb2bb2a817027f1c94bc08ecaa362/datasync/binary_db.go#L39-L63) and [dbTabularData](https://github.com/viamrobotics/app/blob/4e313dba532bb2bb2a817027f1c94bc08ecaa362/datasync/tabular_db.go#L36-L59)

File: proto/viam/app/data/v1/data.proto
Comment (agreenb) [inline proto/viam/app/data/v1/data.proto:62]: Removed in corresponding app PR. We were otherwise always returning STATUS_SUCCESS since we changed our deletion API to not support partial deletion.

File: proto/viam/app/model/v1/model.proto
Comment (agreenb) [inline proto/viam/app/model/v1/model.proto:1]: Replaced months ago in both RDK and app by packages.proto

File: proto/viam/app/packages/v1/packages.proto
Comment (agreenb) [inline proto/viam/app/packages/v1/packages.proto:95]: Replaced by [dbPackage](https://github.com/viamrobotics/app/blob/4e313dba532bb2bb2a817027f1c94bc08ecaa362/packages/db.go#L76-L90).

**Comment by agreenb** [comment]
Both reviewers are out but will be back Monday. Since this is breaking, will add the make dist/buf autogenerated files after reviews from you both and before getting approval/merge by Eric.

File: proto/viam/app/packages/v1/packages.proto
Comment (tahiyasalam) [reply proto/viam/app/packages/v1/packages.proto:95]: Now that we've removed `InternalPackage`, seems like we need to remove the import "tagger/v1/tagger.proto"

File: proto/viam/app/datasync/v1/data_sync.proto
Comment (AaronCasas) [inline proto/viam/app/datasync/v1/data_sync.proto:59]: Should session_id be removed too? I don't think it's ever been used. I know not in the scope though - so maybe should save for some other time

**Review by tahiyasalam — APPROVED** [review]
LGTM!

**Review by AaronCasas — APPROVED** [review]
LGTM!

**Pull request merged**
